cpp/include/cudf/detail/stream_compaction.hpp

```diff
@@ -82,10 +82,33 @@ std::unique_ptr<table> unique(
 std::unique_ptr<table> distinct(
   table_view const& input,
   std::vector<size_type> const& keys,
+  duplicate_keep_option keep          = duplicate_keep_option::KEEP_ANY,
   null_equality nulls_equal           = null_equality::EQUAL,
   rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Get all the row indices of the input table such that the rows corresponding to these
+ *        indices are distinct from each other.
+ *
+ * Given an `input` table_view, an output array of row indices is generated by adding as many
+ * indices as possible, in an unspecified order, such that all rows correspond to these indices
+ * are distinct (i.e., an arbitrary pair of rows are never be compared as equivalent).
+ *
+ * @param input The input table
+ * @param keep Get index of the first, last, any, or none row among the found duplicates rows
+ * @param nulls_equal Flag to specify whether null elements should be considered as equal
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned vector
+ * @return A device_uvector containing the result indices
```

Comment:
@bdice May I ask your help to polish the doc here since you reviewed my previous stream compaction refactor?

Reply:
I took a pass of review with emphasis on docs. Thanks!